test/Chainweb/Test/Pact/PactSingleChainTest.hs

```diff
@@ -531,6 +533,131 @@ compactionIsIdempotent rdb =
+      let payloadDb = _bdbPayloadDb blockDb
+      sqlEnv <- sqlEnvIO
+      (mempoolRef, mempool) <- do
+        (ref, nonRef) <- dm
+        pure (pure ref, nonRef)
+      pactQueue <- newPactQueue 2000
+
+      let -- creating a module uses about 60k gas. this is
+          -- that plus some change.
+          gasLimit :: GasLimit
+          gasLimit = 70_000
+
+      let ver = testVersion
+      let cfg = testPactServiceConfig {
+            _pactBlockGasLimit = gasLimit
+          }
+      let logger = genericLogger System.LogLevel.Error (\_ -> return ())
+
+      void $ forkIO $ runPactService ver cid logger pactQueue mempool bhDb payloadDb sqlEnv cfg
+
+      let numBlocks :: Num a => a
+          numBlocks = 100
+      let halfwayPoint :: Integral a => a
+          halfwayPoint = numBlocks `div` 2
+
+      setOneShotMempool mempoolRef goldenMemPool
+
+      let createTable :: Int -> Text -> IO ChainwebTransaction
+          createTable n tblName = do
+            let tx = T.unlines
+                  [ "(namespace 'free)"
+                  , "(module m" <> sshow n <> " G"
+                  , "  \"Hullaballo\""
+                  , "  (defcap G () true)"
+                  , "  (defschema empty-schema)"
+                  , "  (deftable " <> tblName <> ":{empty-schema})"
+                  , ")"
+                  , "(create-table " <> tblName <> ")"
+                  ]
+            buildCwCmd
+              $ signSender00
+              $ set cbGasLimit gasLimit
+              $ mkCmd ("createTable-" <> tblName <> "-" <> sshow n)
+              $ mkExec tx
+              $ mkKeySetData "sender00" [sender00]
+
+      let beforeTable = "test_before"
+      let afterTable = "test_after"
+
+      supply <- newIORef @Int 0
+      madeBeforeTable <- newIORef @Bool False
+      madeAfterTable <- newIORef @Bool False
+      replicateM_ numBlocks $ do
+        setMempool mempoolRef $ mempty {
+          mpaGetBlock = \_ _ mBlockHeight _ _ -> do
+            let mkTable madeRef tbl = do
+                  madeYet <- readIORef madeRef
+                  if madeYet
+                  then do
+                    pure mempty
+                  else do
+                    n <- atomicModifyIORef' supply $ \a -> (a + 1, a)
+                    tx <- createTable n tbl
+                    writeIORef madeRef True
+                    pure (V.fromList [tx])
+
+            if mBlockHeight <= halfwayPoint
+            then do
+              mkTable madeBeforeTable beforeTable
+            else do
+              mkTable madeAfterTable afterTable
+        }
+        void $ runBlock pactQueue blockDb second pat
+
+      let db = _sConn sqlEnv
+
+      let compact h = C.withDefaultLogger System.Logger.Types.Error $ \cLogger -> do
+            let flags = [C.Flag_NoVacuum, C.Flag_NoGrandHash]
+            void $ C.compact h cLogger db flags
+
+      let compactionHeight = BlockHeight halfwayPoint
```

Comment:
Let's move the bindings closer to the actual usage? So it would be easier to see what's happening.